test/test_mkldnn_fusion.py

```diff
@@ -246,5 +255,40 @@ def forward(self, x):
                     self.assertEqual(ref, fused)
 
 
+    def test_conv_binary_fusion_ops(self):
+        class M(nn.Module):
+            def __init__(self, binary_fn, dim, in_channels, out_channels, dilation, groups, bias, **kwargs):
+                super(M, self).__init__()
+                self.conv = CONV_MODULES[dim](in_channels, out_channels, dilation=dilation, groups=groups, bias=bias, **kwargs)
+                self.binary = binary_fn
+
+            def forward(self, x, other):
+                x = self.conv(x)
+                x = self.binary(x, other)
+                return x
+
+        input_shapes = {2: (112, 112), 3: (55, 55, 55)}
+        for pointwise_name, pointwise_fn in self._binary_list().items():
+            for dim in [2, 3]:
+                channels_last = torch.channels_last if dim == 2 else torch.channels_last_3d
+                options = itertools.product([True, False], [1, 2], [1, 4], [torch.contiguous_format, channels_last])
+                for bias, dilation, groups, memory_format in options:
+                    oC = 10 * groups
+                    iC = 3 * groups
+                    x_shape = (1, iC) + input_shapes[dim]
+                    x = torch.randn(x_shape, dtype=torch.float32).to(memory_format=memory_format)
+                    mod = M(pointwise_fn, dim, iC, oC, dilation, groups, bias, kernel_size=3)
+                    mod = mod.to(memory_format=memory_format).eval()
+                    other = torch.randn_like(mod.conv(x))
```

Comment:
Test same dtype and different dtype?

Reply:
Now, I add a dtype check which makes sure all input should have same type. and for mixed-dtype, we will report an error to user.